src/engine/cache/FileCache.class.js

```diff
@@ -117,18 +102,29 @@ class FileCache extends BaseCache {
     const cachePath = path.join(this.fileFolder, cacheFilename)
 
     await fs.copyFile(filePath, cachePath)
-
-    databaseService.saveFile(this.filesDatabase, timestamp, cachePath)
     this.logger.debug(`File "${filePath}" cached in "${cachePath}".`)
   }
 
   /**
    * Retrieve files from the Cache database and send them to the associated northHandleFileFunction function
    * This method is called when the group count is reached or when the Cache timeout is reached
-   * @returns {{path: string, timestamp: number}|null} - The file to send
+   * @returns {Promise<{path: string, timestamp: number}|null>} - The file to send
    */
-  retrieveFileFromCache() {
-    return databaseService.getFileToSend(this.filesDatabase)
+  async retrieveFileFromCache() {
+    const fileNames = await fs.readdir(this.fileFolder)
+
+    if (fileNames.length === 0) {
+      return null
+    }
+
+    const sortedFiles = fileNames
+      .map(async (fileName) => ({
+        path: `${this.fileFolder}/${fileName}`,
+        timestamp: (await fs.stat(`${this.fileFolder}/${fileName}`)).mtime.getTime(),
+      }))
+      .sort((a, b) => a.time - b.time)
```

Comment:
This is what we used originally to generate the timestamp: https://github.com/OptimistikSAS/OIBus/blob/main/src/engine/cache/FileCache.class.js#L112
And `mTime` is a Date object: https://nodejs.org/api/fs.html#statsmtime

Reply:
My comment was about the sorting (l125): the map function outputs an object with path and timestamp, not time.

Also, it may be better to use path.resolve(this.fileFolder, fileName) instead of `${this.fileFolder}/${fileName}` to be cross platform compatible

Reply:
Ahh, yeah. Fixed it. And also added a migration to remove old file cache DBs.